src/v/config/bounded_property.h

```diff
@@ -111,7 +140,48 @@ struct numeric_bounds {
     }
 };
 
-template<typename T, typename I = typename detail::inner_type<T>::inner>
+/**
+ * Define valid bounds for any numeric configuration property.
+ */
+template<detail::numeric T>
+struct numeric_bounds {
```

Comment:
I have considered that option too. The other plan was to `if constexpr` the alignment and odd/even functions based on whether the underlying type supports `%` or not. The reason I have selected this way is this. When `bound_property<double,numeric_bounds>` is constructed now, you will get an error if you try to initialize numeric bounds with `.align` or `.oddeven` members in the list. With the alternate approach, it is still possible to have something like
```
  , kafka_memory_share_for_fetch(
      *this,
      "kafka_memory_share_for_fetch",
      "",
      {.needs_restart = needs_restart::yes, .visibility = visibility::user},
      0.3,
      {.min = 0.0, .max = 1.0, .align={}, .oddeven={}})
```
That is okish since it is still not possible to supply actual values to `.align' and `.oddeven`, but it's better when you can't even mention them in that context.

Another reason is that this approach also creates a framework to use other "something_else_bounds".

LMK what you think of that, as I'm still unsure which way is better.